app/controllers/users_controller.rb

```diff
@@ -7,24 +7,13 @@ def new
 
   def create
     @user = User.new(user_params)
-    if @user.valid?
-      charge = StripeWrapper::Charge.create(
-          amount: 999, # amount in cents, again
-          source: params[:stripeToken],
-          description: "Sign up charge for #{@user.email}"
-        )
-      if charge.successful? 
-        @user.save
-        handle_invitation
-        AppMailer.delay.send_welcome_email(@user)
-        flash[:success] = "Thank you for registering with MyFlix. Please Sign in now."
-        redirect_to sign_in_path
-      else
-        flash.now[:error] = charge.error_message
-        render :new
-      end
+    result = UserSignup.new(@user).sign_up(params[:stripeToken], params[:invitation_token])
+
+    if result.successful?
+      flash[:success] = "Thank you for registering with MyFlix. Please Sign in now."
+      redirect_to sign_in_path
     else
-      flash.now[:error] = "Invalid user information. Please check the errors below."
+      flash[:error] = result.error_message
```

Comment:
Always `flash.now` on render. 

HOMEWORK: Explain why flash.now is the right way to go on render.